### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by simonbyrne** [JuliaMath/SpecialFunctions.jl#159]
The documentation of this package leaves a lot to be desired. Some ideas:

- Split it up into obvious sections (perhaps along the lines of the [DLMF](https://dlmf.nist.gov/))
- Document the domains/where the function is defined
- Add external links (Wikipedia and DLMF are the obvious ones to link to).
- References to implementations
- Add plots
  - handle discontinuities nicely
  - nice plotting of complex functions (see https://reference.wolfram.com/language/ref/ComplexPlot3D.html for some ideas)

**Comment by Sumegh-git** [comment]
I'll do some of this work this weekend if not earlier. Also where should these changes be done? The domains and ext. links can be done in the function docstring where they are defined. What about plots?

**Comment by simonbyrne** [comment]
I don't think it is possible to have docs in docstrings, they would have to go into the markdown files. I have some code I'll post.

**Comment by PaulXiCao** [comment]
I'd like to help out here. 
Have you got any special layout in mind? I'd just create sections along DLMF and then a subsection for each function (sorted alphabetically).
Each function would be described in the same manner: domain+links(wikipedia/DLMF)+plots [optinally: implementation details+todos].

What are thoughts on it?

**Comment by simonbyrne** [comment]
@PaulXiCao That would be fantastic. I would encourage you to start with small incremental changes and we can go from there.

**Comment by PaulXiCao** [comment]
@simonbyrne I just uploaded the first two documentations about `erf`, and `erfc`. I'd be happy for some more input.

My thoughts this far:

1. Section names according to DLMF but no further subsectioning except one for each function. For me this seems enough at first .
2. I didnt add any more explanations or plots bc i think nobody really cares. Doesnt just everybody go to wikipedia for further informations?
3. What do you think about the layout? Very basic but is it enough or did you have something fancier in mind?
